assets/src/settings-page/index.js

```diff
@@ -127,24 +127,29 @@ function Root() {
 	const { downloadingTheme } = useContext( ReaderThemes );
 	const [ saved, setSaved ] = useState( false );
 
+	// Show the success notice after options have saved.
+	useEffect( () => {
+		if ( didSaveOptions ) {
+			setSaved( true );
+		}
+	}, [ didSaveOptions ] );
+
 	/**
 	 * Shows a saved notice on success.
 	 */
 	useEffect( () => {
-		if ( true === didSaveOptions && ! downloadingTheme ) {
-			setSaved( true );
-
-			const timeout = setTimeout( () => [
-				setSaved( false ),
-			], 9000 );
+		if ( saved && ! downloadingTheme ) {
+			const timeout = setTimeout( () => {
+				setSaved( false );
```

Comment:
The `saved` state name seems a bit misleading here. It's not like after 9 seconds the form then enters a dirty state.  Wouldn't it be more clear to call this like `showingSavedNotice` or something?

Reply:
I took a slightly different approach in the latest commit, e3406e391024945bcb200b36d75615a1363eea5e.

Instead of showing/hiding the success notice with the boolean, we'll use CSS classes. This prevents me from having to add additional state to manage the show/hide CSS animations.

The way it works with this update is:

1. Instead of the `saved` state, we have `savedNoticeClass` which is initially empty, and the notice is hidden offscreen via a CSS transform.
2. When `savedNoticeClass` is `visible`, the notice slides in.
3. When `savedNoticeClass` is `dismissed`, the notice slides back out.
4. If `savedNoticeClass` is `visible` and the user changes something, the class is changed to `dismissed`. 
5. If `savedNoticeClass` is `visible` for 9 seconds, the class is changed to `dismissed`.